src/parse.d

```diff
@@ -2867,7 +2867,9 @@ final class Parser : Lexer
                             error("scope cannot be ref or out");
 
                         Token* t;
-                        if (tpl && token.value == TOKidentifier && (t = peek(&token), (t.value == TOKcomma || t.value == TOKrparen || t.value == TOKdotdotdot)))
+                        if (tpl && token.value == TOKidentifier
+                            && (t = peek(&token)) !is null
```

Comment:
`t` can never be `null`, so testing for it is misleading. It would be better to just divide the expression into two `if` statements.